app/core/views.py

```diff
@@ -87,26 +88,70 @@ def edit_profile():
 def follow(username):
     user = User.query.filter_by(username=username).first()
     if not user:
-        flash("User not found")
-        return redirect(url_for("core.index"))
+        response = Response(
+            response=json.dumps({
+                'error': 'User not found'
+            }),
+            status=502,
+            mimetype='application/json'
+        )
+        return response
+
     if user == current_user:
-        flash("You can't follow yourself")
-        return redirect(url_for('core.user', username=username))
+        response = Response(
+            response=json.dumps({
+                'error': "You can't follow yourself"
+            }),
+            status=502,
+            mimetype='application/json'
+        )
+        return response
+
     current_user.follow(user)
     db.session.commit()
-    flash(f"You are following {username}!")
-    return redirect(url_for("core.user", username=username))
+
+    response = Response(
+        response=json.dumps({
+            'error': ""
+        }),
+        status=200,
+        mimetype='application/json'
+    )
+
+    return response
 
 @bp.route('/unfollow/<username>')
 def unfollow(username):
     user = User.query.filter_by(username=username).first()
     if not user:
-        flash("User not found")
-        return redirect(url_for("core.index"))
+        response = Response(
+            response=json.dumps({
+                'error': 'User not found'
+            }),
+            status=502,
+            mimetype='application/json'
+        )
+        return response
+
     if user == current_user:
-        flash("You can unfollow yourself")
-        return redirect(url_for("core.user", username=username))
+        response = Response(
+            response=json.dumps({
+                'error': "You can't unfollow yourself"
+            }),
+            status=502,
+            mimetype='application/json'
+        )
+        return response
+
     current_user.unfollow(user)
     db.session.commit()
-    flash(f"You unfollowed {username}")
-    return redirect(url_for('core.user', username=username))
\ No newline at end of file
+
+    response = Response(
```

Comment:
Please note that, **follow**, **unfollow** is broken in **user popup**. Please hover over some user profile links to see the user popup.
![userpopup](https://user-images.githubusercontent.com/24582910/66103954-d7ed6e80-e5d4-11e9-8f21-0a951187b742.png)